spectroscopy/code_src/Euclid_functions.py

```
@@ -0,0 +1,185 @@
+
+    This function performs a cone search on the Euclid MER catalog via IRSA (a cloud-based service),
+    retrieves associated 1D spectral file URIs from a TAP-accessible association table,
+    downloads the spectra FITS files, extracts flux and error, and packages them
+    into a MultiIndexDFObject.
+
+    Parameters
+    ----------
+    sample_table : list of dict
+        A list of source dictionaries, each containing:
+        - 'label' : str
+            A unique identifier for the source (used for indexing).
+        - 'coord' : astropy.coordinates.SkyCoord
+            The sky coordinates of the source.
+
+    search_radius_arcsec : float
+        Search radius (in arcseconds) for the initial cone search around each coordinate.
+
+    Returns
+    -------
+    df_spec : MultiIndexDFObject
+        A structured multi-indexed DataFrame containing:
+        - wave : Quantity array
+        - flux : Quantity array
+        - err : Quantity array
+        along with metadata fields ('label', 'objectid', 'mission', 'instrument', 'filter').
+
+    Notes
+    -----
+    - This function accesses IRSA services and downloads FITS files from the cloud.
+    - It assumes valid internet connectivity and may fail or be slow under network issues.
+    - The function suppresses Astropy warnings during parsing.
+    """
+    df_spec = MultiIndexDFObject()
+    table_mer = "euclid_q1_mer_catalogue"
+    table_1dspectra = "euclid.objectid_spectrafile_association_q1"
+
+    for stab in sample_table:
+        print(f"Processing source {stab['label']}")
+        coord = stab["coord"]
+
+        # Step 1: Cone search in MER catalog
+        try:
+            tab = Irsa.query_region(
+                coordinates=coord,
+                catalog=table_mer,
+                spatial="Cone",
+                radius=search_radius_arcsec * u.arcsec
+            )
+        except Exception as e:
+            print(f"Query failed for {stab['label']}: {e}")
+            continue
```

Don't swallow error, and practically never do an `except Exception`, it is just in very rare cases the appropriate option.

---

This comment stands for all the remaining try/excepts, try to remove them, and definitely change the exception type for the ones remaining

---

I made the errors more explicit let me know if it still is not ok. Or go ahead and fix if you prefer, thanks.